setup.py

```diff
@@ -997,6 +1027,7 @@ def make_relative_rpath(path):
         package_data={
             'torch': [
                 'lib/*.so*',
+                'lib/*.a',
```

Comment:
We need to have access to libprotobuf.a here. Could just add those to this list, but seemed to make more sense to just bring all .a over.

Reply:
why are we shipping any .a members? (even libprotobuf.a), can you explain / add a comment?

Reply:
At least for protobuf, statically linking the library with hidden visibility was the best way we could avoid protobuf version mismatch problens and gcc-ABI incompatibility problems

Reply:
libprotobuf.a is built and included as hidden in libcaffe2.so. That means that any other user of protobuf's exposed from there (libtorch.so, _C, etc.) and even outside parties will need to link against it. However, this might not be needed... I did a lot of iteration on this PR, so I should verify first. Sounds like we should limit to libprotobuf.a if at all?

Reply:
i am not questioning why we are using libprotobuf.a, seems great.

Shipping it into our {wheel file, conda package} etc. seems to be unnecessary, as the symbols are already in libcaffe2.so.

We should link it into libcaffe2.so with a whole-archive flag so that all symbols (not just ones caffe2 is using) get linked, which means any downstream library that wants to use protobuf capabilities can just link against libcaffe2.so and get full resolution of all protobuf symbols it needs.

For example any extension, or libtorch, or whatever, if they need libprotobuf.a symbols, they can get by simply linking to libcaffe2.so